databricks/koalas/indexes.py

```diff
@@ -958,6 +958,56 @@ def max(self):
 
         return result if len(result) > 1 else result[0]
 
+    def append(self, other):
+        """
+        Append a collection of Index options together.
+
+        Parameters
+        ----------
+        other : Index
+
+        Returns
+        -------
+        appended : Index
+
+        Examples
+        --------
+        >>> kidx = ks.Index([10, 5, 0, 5, 10, 5, 0, 10])
+        >>> kidx
+        Int64Index([10, 5, 0, 5, 10, 5, 0, 10], dtype='int64')
+
+        >>> kidx.append(kidx)
+        Int64Index([10, 5, 0, 5, 10, 5, 0, 10, 10, 5, 0, 5, 10, 5, 0, 10], dtype='int64')
+
+        Support for MiltiIndex
+
+        >>> kidx = ks.MultiIndex.from_tuples([('a', 'x'), ('b', 'y')])
+        >>> kidx  # doctest: +SKIP
+        MultiIndex([('a', 'x'),
+                    ('b', 'y')],
+                   )
+
+        >>> kidx.append(kidx)  # doctest: +SKIP
+        MultiIndex([('a', 'x'),
+                    ('b', 'y'),
+                    ('a', 'x'),
+                    ('b', 'y')],
+                   )
+        """
+        if type(self) is not type(other):
+            raise NotImplementedError(
+                "append() between Index & MultiIndex currently is not supported")
+
+        sdf_self = self._internal.sdf
+        sdf_other = other._internal.sdf
+        sdf_appended = sdf_self.union(sdf_other)
```

Comment:
@itholic can you also check when the name of indexes are different, and when they are index / multi-index?

Reply:
> can you also check when the name of indexes are different

added test cases & fix related logic

> and when they are index / multi-index?

i think we currently hard to support since we can't make index with various types like below pandas' case.

```python
>>> pidx1 = pd.Index([10, 5, 0, 5, 10, 5, 0, 10])
>>> pidx2 = pd.MultiIndex.from_tuples([('a', 'x'), ('b', 'y')])
>>> pidx1.append(pidx2)
Index([10, 5, 0, 5, 10, 5, 0, 10, ('a', 'x'), ('b', 'y')], dtype='object')
```

Reply:
> pandas doesn't keep the name of index, so we're also following that behavior

Actually pandas does keep the name of index for multi-index.

```py
>>> pmidx1
MultiIndex([('a', 'x', 1),
            ('b', 'y', 2),
            ('c', 'z', 3)],
           names=['x', 'y', 'z'])
>>> pmidx2
MultiIndex([('a', 'x', 1),
            ('b', 'y', 2),
            ('c', 'z', 3)],
           names=['p', 'q', 'r'])
>>> pmidx1.append(pmidx2)
MultiIndex([('a', 'x', 1),
            ('b', 'y', 2),
            ('c', 'z', 3),
            ('a', 'x', 1),
            ('b', 'y', 2),
            ('c', 'z', 3)],
           names=['x', 'y', 'z'])
```

Reply:
@ueshin 

opps i don't understand why they only keep the `names` for `MultiIndex` while not `name` for `Index` ! (just joke i'll manage it 😄 )

Thanks for checking this out!!

Reply:
> i don't understand why they only keep the `names` for `MultiIndex` while not keep `name` for `Index` !

Agree. I prefer to keep the name even for `Index` .. Let's follow pandas for now. 😺

Reply:
@ueshin 

yeah let's just keep following pandas 👍  

i just fixed it :)